File: Challenger.Domain/Dtos/DishDto.cs
using Challenger.Domain.Food;
using System.Collections.Generic;

namespace Challenger.Domain.Dtos
{
    public class DishDto : INutrients
    {
        public long Id { get; set; }

        public long UserId { get; set; }

        public List<IngridientDto> Ingridients { get; set; }

        public string Name { get; set; }

        public bool IsPublic { get; set; }

        public int PreparationTime { get; set; }

        public int Servings { get; set; }

        public double Energy { get; set; }

        public double Fats { get; set; }

        public double Proteins { get; set; }

        public double Carbohydrates { get; set; }
    }
}

File: Challenger.Web/src/app/components/food/dish-add/DishDto.ts
import { IngridientDto } from "../ingridient-add/IngridientDto";

export class DishDto {

    id: number;
    userId: number;
    name: string;
    ingridients: IngridientDto[]
    isPublic: boolean;
    preparationTime: number;
    servings: number;
    size: number;
    energy: number;
    fats: number;
    proteins: number;
    carbohydrates: number;
}